confgenerator/agentmetrics.go

```diff
@@ -28,11 +27,192 @@ import (
+	"UNKNOWN":             "500",
+	"INTERNAL":            "500",
+	"DATA_LOSS":           "500",
+	"UNIMPLEMENTED":       "501",
+	"UNAVAILABLE":         "503",
+	"DEADLINE_EXCEEDED":   "504",
+}
+
+func (r AgentSelfMetrics) otelPipelineProcessorsWithOtelLogging() map[string][]otel.Component {
+	return map[string][]otel.Component{"metrics": {
+		otel.MetricsFilter(
+			"include",
+			"strict",
+			"otelcol_process_uptime",
+			"otelcol_process_memory_rss",
+			"otelcol_exporter_sent_log_records",
+			"otelcol_exporter_send_failed_log_records",
+			"grpc.client.attempt.duration",
+			"googlecloudmonitoring/point_count",
+		),
+		otel.Transform("metric", "metric",
+			[]ottl.Statement{
+				// Make two copies of request duration for logging and monitoring
+				ottl.CopyMetric("grpc.client.attempt.duration.logging", `grpc.client.attempt.duration`),
+				ottl.CopyMetric("grpc.client.attempt.duration.monitoring", `grpc.client.attempt.duration`),
+				// Extract count from histogram metrics
+				ottl.ExtractCountMetric(true, "grpc.client.attempt.duration.logging"),
+				ottl.ExtractCountMetric(true, "grpc.client.attempt.duration.monitoring"),
+			},
+		),
+		otel.MetricsOTTLFilter([]string{}, []string{
+			// Filter out histogram datapoints where the grpc.target is not related.
+			`metric.name == "grpc.client.attempt.duration.logging_count" and (not IsMatch(datapoint.attributes["grpc.target"], "logging.googleapis"))`,
+			`metric.name == "grpc.client.attempt.duration.monitoring_count" and (not IsMatch(datapoint.attributes["grpc.target"], "monitoring.googleapis"))`,
+		}),
+		otel.MetricsFilter(
+			"include",
+			"strict",
+			"otelcol_process_uptime",
+			"otelcol_process_memory_rss",
+			"otelcol_exporter_sent_log_records",
+			"otelcol_exporter_send_failed_log_records",
+			"grpc.client.attempt.duration.logging_count",
+			"grpc.client.attempt.duration.monitoring_count",
+			"googlecloudmonitoring/point_count",
+		),
+		otel.MetricsTransform(
+			otel.DuplicateMetric("otelcol_exporter_send_failed_log_records", "agent/log_entry_retry_count",
+				// change data type from double -> int64
+				otel.ToggleScalarDataType,
+				otel.AddLabel("response_code", "400"),
+				otel.AggregateLabels("sum", "response_code"),
+			),
+			otel.RenameMetric("otelcol_exporter_sent_log_records", "agent/log_entry_count",
+				// change data type from double -> int64
+				otel.ToggleScalarDataType,
+				otel.AddLabel("response_code", "200"),
+				otel.AggregateLabels("sum", "response_code"),
+			),
+			otel.RenameMetric("otelcol_exporter_send_failed_log_records", "agent/log_entry_count",
+				// change data type from double -> int64
+				otel.ToggleScalarDataType,
+				otel.AddLabel("response_code", "400"),
+				otel.AggregateLabels("sum", "response_code"),
+			),
+			otel.DuplicateMetric("otelcol_process_uptime", "agent/uptime",
+				// change data type from double -> int64
+				otel.ToggleScalarDataType,
+				otel.AddLabel("version", r.LoggingVersionLabel),
```

Comment:
I don't think we should be producing a second agent uptime stream. For better or worse, the otel collector is the "metrics" agent even though it already handles metrics and traces and now we're adding logging. The only other uptime metric we should have is the one from fluent-bit.

Reply:
The only consideration of not having an `agent/uptime` with `version: logging` is that we would need modified version of the integration tests when testing both at the same time. This eventually can be removed, but keeping both would make sure it generates exactly the same set of metrics as fluent-bit does.

Reply:
Ok! I implemented this, now fluent-bit `uptime` is reported as is.